viewing_party/party.py

```diff
@@ -1,23 +1,175 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if title is None:
+        return None
+    if genre is None:
+        return None
+    if rating is None:
+        return None
+    
+    movie_dict["title"] = title
+    movie_dict["genre"] = genre
+    movie_dict["rating"] = rating
+
+    return movie_dict
+
+
+def add_to_watched(user_data, movie):
+    watched = user_data["watched"] 
+    if watched is None:        
+        watched = [movie]
+    watched.append(movie)
+
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    watchlist = user_data["watchlist"]
+    if watchlist is None:
+        watchlist= [movie]
+    watchlist.append(movie)
+
+    return user_data
+
+        
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data  
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    average = 0.0
+    total = len(user_data["watched"])
+
+    for movies in user_data["watched"]:
+        for key, value in movies.items():
+            if key == "rating":
+                sum += value
+            average = sum /total
+    return average
+                        
+
+def get_most_watched_genre(user_data):
+    # create frequency dict
+    frequency = {}
+
+    if not user_data["watched"]:
+        return None
+    
+    for movies in user_data["watched"]:

```

Comment:
`user_data["watched"]` is a list of movies that a user has watched. When you iterate over the list, each element is a single movie. 

A more accurate variable name for each element in this list would be "movie", instead of "movies".
```suggestion
    for movie in user_data["watched"]:
```